opencog/atoms/atom_types/atom_types.script

```diff
@@ -188,6 +183,13 @@ NOT_LINK <- UNORDERED_LINK,CRISP_OUTPUT_LINK,BOOLEAN_LINK
 SEQUENTIAL_AND_LINK <- ORDERED_LINK,CRISP_OUTPUT_LINK,BOOLEAN_LINK
 SEQUENTIAL_OR_LINK <- ORDERED_LINK,CRISP_OUTPUT_LINK,BOOLEAN_LINK
 
+// Measure-theoretic (probabilistic) versions of AND_LINK, OR_LINK
+// These use probabilistic formulas to compute truth values.
+// See issue opencog/atomspace#2814 for a discussion.
+UNION_LINK <- OR_LINK
+INTERSECTION_LINK <- AND_LINK
+COMPLEMENT_LINK <- NOT_LINK
```

Comment:
Cool!  I'm not actually sure they should inherit OrLink, etc.  But I guess we can revisit that later on.

Reply:
Yes, you are absolutely right... this was a rush job, I didn't give it any thought.  The reason they should not is because ... the `checkers.cc` will notice that these inherit evaluatability, and thus will prevent you  from saying `(Union (Set (Concept "foo")))` Dohhh!  So this needs to be patched! Which is what #2814 was all about...

It's past bedtime for me; you have any ideas of the correct inheritance for this?

Reply:
Seems like UnorderedLink is the most they should inherit from...!?

Reply:
#2844 fixes this. Review and merge if it looks right.